### PR TITLE
Add RandomResizedCrop preprocessing layer

### LINKED ISSUE #21822: Add RandomResizedCrop
https://github.com/keras-team/keras-cv/issues/131

### PR DESCRIPTION
**Pull request by MalyalaKarthik66**
This PR adds `keras.layers.RandomResizedCrop`, a preprocessing layer that
randomly crops and resizes images to a fixed target size.

The layer:
- samples random crop area and aspect ratio during training
- applies a deterministic center crop during inference
- supports batched image inputs across all backends

This implementation supports image tensors only. For bounding boxes and
segmentation masks, users can rely on `keras_cv.layers.RandomResizedCrop`.

Closes #21822

### CODE REVIEW

**Comment by MalyalaKarthik66** [comment]
@hertschuh  Gentle reminder on this PR when convenient.

There is still one CI failure on the Torch backend, although the same test passes locally. I may be missing something subtle here—could you please take a look when you have time? Thanks!

**Review by hertschuh — COMMENTED** [review]
First sorry for the delay on this.

For consistency with all the other image preprocessing layers, this should also support bounding boxes and masks.

Note that once this PR is in: https://github.com/keras-team/keras/pull/22192

It will be quite easy to add support for masks.

File: keras/src/layers/preprocessing/image_preprocessing/random_resized_crop.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        if transformation is None:
+            transformation = self.get_random_transformation(images, training)
+        h, w, ch, cw = transformation
+        images = self._slice_images(images, h, w, ch, cw)
+        return backend.image.resize(
+            images,
+            size=(self.height, self.width),
+            interpolation=self.interpolation,
+            data_format=self.data_format,
+        )
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/image_preprocessing/random_resized_crop.py]: If you rebase to master to get https://github.com/keras-team/keras/pull/22192 then you can rename this to:
```python
def _transform_images(self, images, transformation, interpolation):
```
And you'll get support for masks automatically. But you still need to support bounding boxes.

File: keras/src/layers/preprocessing/image_preprocessing/random_resized_crop_test.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+SKIP_BACKENDS = ["openvino"]
+
+pytestmark = pytest.mark.skipif(
+    backend.backend() in SKIP_BACKENDS,
+    reason=(
+        "RandomResizedCrop tests not supported for backend: {}".format(
+            backend.backend()
+        )
+    ),
+)
```
Comment (hertschuh) [inline keras/src/layers/preprocessing/image_preprocessing/random_resized_crop_test.py:18]: Instead of this, please add a line `RandomResizedCropTest` to this file: https://github.com/keras-team/keras/blob/master/keras/src/backend/openvino/excluded_concrete_tests.txt#L148

**Review by hertschuh — COMMENTED** [review]
Sorry again for the delay.

Also, the tests are failing on Torch. The tests need to work with `channels_first` instead of `channels_last`. You can look at how the other tests do this.
